### PR TITLE
[prometheusreceiver and prometheusremotewriteexporter] Update resource attributes used when translating to/from prometheus

### LINKED ISSUE #7081: Scrapers - Capture host and port of monitored endpoint as resource attributes
Metric scrapers typically pull data from a known host and port. The metrics produced by a scraper would be more useful if placed in the context of the resource they describe. Therefore, scrapers should capture these values as resource attributes, when possible. 

The semantic conventions define the attributes `net.host.name` and `net.host.port` [here](https://github.com/open-telemetry/opentelemetry-specification/blob/e037c49ed77c043fb9715852ca5ee6e4ed1bf391/specification/trace/semantic_conventions/span-general.md#general-network-connection-attributes), and I believe these are appropriate for this use case.

**Additional Context**

Prometheus produces telemetry with basically the same information via the [instance label](https://prometheus.io/docs/concepts/jobs_instances/), although this is a combined field, defined as `<host>:<port> part of the target's URL that was scraped`. 

The `prometheusreceiver` [splits the `instance` and saves the host and port as `host.name` and `port`](https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/main/receiver/prometheusreceiver/internal/prom_to_otlp.go#L44-L61). We likely want to use the same attribute names in both the prometheus and non-prometheus cases, but perhaps some prometheus experts can shed light on whether there is a distinction here I'm missing.

### LINKED ISSUE #8264: [prometheusreceiver]/[prometheusremotewriteexporter] Update resource attributes to reflect the updated prom spec
After https://github.com/open-telemetry/opentelemetry-specification/pull/2381, we should generate the following resource attributes in the prometheus receiver, and prometheus remote write exporter

Changed:
* `instance` -> `service.instance.id`
* `host.name` -> `net.host.name`
* `port` -> `net.host.port`
* `scheme` -> `http.scheme`

Removed: `job`

Additionally, when reconstructing `job` in prometheus exporters, use `<service.namespace>/<service.name>`

### PR DESCRIPTION
**Pull request by dashpole**
**Description:** 

Fixes https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/8264.  
Fixes https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/7081.

This PR updates the resource attributes used when translating to and from prometheus to match the [prometheus conversion specification](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/metrics/datamodel.md#prometheus-compatibility), which was updated in https://github.com/open-telemetry/opentelemetry-specification/pull/2381 with these attributes.

**Testing:** Unit tests were updated.  Prometheus compatibility tests still pass after this change.

### CODE REVIEW

**Review by Aneurysm9 — APPROVED** [review]

File: receiver/prometheusreceiver/internal/prom_to_otlp.go [hunk 1]
```diff
@@ -50,12 +50,11 @@ func CreateNodeAndResourcePdata(job, instance, scheme string) *pdata.Resource {
 	attrs := resource.Attributes()
 	attrs.UpsertString(conventions.AttributeServiceName, job)
 	if isDiscernibleHost(host) {
-		attrs.UpsertString(conventions.AttributeHostName, host)
+		attrs.UpsertString(conventions.AttributeNetHostName, host)
```
Comment (codeboten) [inline receiver/prometheusreceiver/internal/prom_to_otlp.go:53]: does this require a breaking change note in the changelog?

Comment (dashpole) [reply receiver/prometheusreceiver/internal/prom_to_otlp.go:53]: Yes.  Right now, i've covered this with:

`prometheusreceiver` Produces different resource attributes when converting metrics to OTLP (#8266)

Do you have a different suggestion?

Comment (codeboten) [reply receiver/prometheusreceiver/internal/prom_to_otlp.go:53]: I'd suggest an item in the breaking changes section calling out the rename from `host.name` to `net.host.name` to make it explicit for any users.

Comment (dashpole) [reply receiver/prometheusreceiver/internal/prom_to_otlp.go:53]: updated to explicitly include each resource attribute changed

**Review by codeboten — APPROVED** [review]
Thanks @dashpole

**Pull request merged**